### DISCUSSION TITLE
error while loading shared libraries: libhm_reader_linux64.so

### DISCUSSION THREAD

**Discussion by sezerdeger** [OpenRadioss/OpenRadioss#419] [Users]
Dear all,

I am quite new to the world of Linux (just half a day years old). I am following the guide named "GettingStartedWithOpenRadiossTutorial_2022_09_08" which is posted on the OpenRadioss website. Although I can see the library file named "libhm_reader_linux64.so" under the directory "/mnt/d/WS/OpenRadioss/extlib/hm_reader/linux64/", it cannot adress it somehow when I try to run the following command:

./starter_linux64_gf -i BIRD_WINDSHIELD_v1_0000.rad

For information, I have copied starter_linux64_gf file after compiling to the working directory which also includes .rad files. I have searched the web and found that this problem might be related to the environment variables of windows probably. 

I have also used sudo find libhm_reader_linux64.so command and still no avail. The program says no such file or directory.

Can somebody help me please?

Best,
Sezer Değer

**Comment by panda1100** [comment]
Did you set environment variables? Followings are my environment variables for OpenRadioss, I built OpenRadioss myself so you might need to set different values on your environment.

```
export OPENRADIOSS_PATH=/opt/OpenRadioss
export RAD_CFG_PATH=$OPENRADIOSS_PATH/hm_cfg_files
export OMP_STACKSIZE=400m

export PATH=$OPENRADIOSS_PATH/exec:$PATH
export LD_LIBRARY_PATH=$OPENRADIOSS_PATH/extlib/hm_reader/linux64/:$OPENRADIOSS_PATH/extlib/h3d/lib/linux64/:$LD_LIBRARY_PATH
```

**Reply by sezerdeger** [reply]
Thank you very much for the guidance. I got it now.

For those who got stuck like me, the files with ".md" extension give enough information. The text with the name "INSTALL.md" is about the environment variables I have asked for.

**Reply by PaulAltair** [reply]
Thanks for pointing this out, I have now added a page to that getting started pdf that refers to the environment variables step and references INSTALL.md

In the next weeks, I hope to produce something as a confluence version of the getting started pdf that is a bit neatened up! e.g. I'm running in Virtual Box rather than WSL and would like to show that route too

**Reply by nawijn** [reply]
As an additional comment, if you are new to Linux and need to play around with custom installed/compiled code, it is worth knowing the existence of the `ldd` command. This, when given an executable or shared object file (.so is the equivalent of a Windows DLL), will tell you whether or not all required libraries are found.

For example:
`ldd  engine_linux64_gf` 

Produces the following output:
`	linux-vdso.so.1 (0x00007ffeb4de7000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f0833ca8000)
	librt.so.1 => /lib64/librt.so.1 (0x00007f0833ca3000)
	libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f0833a6f000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f0833991000)
	libgomp.so.1 => /lib64/libgomp.so.1 (0x00007f0833947000)
	libquadmath.so.0 => /lib64/libquadmath.so.0 (0x00007f08338fd000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f08338f8000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f08336f6000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f0833cc7000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f08336d6000)
`
If one ore more of the libraries is not found, `ldd` will tell you.
